bunner-macroquad/src/grass.rs

```diff
@@ -0,0 +1,201 @@
+            };
+
+        let mut children: Vec<Child> = Vec::new();
+        if hedge_row != HedgeRow::None {
+            // See comments in classify_hedge_segment for explanation of previous_mid_segment
+            #[allow(unused_assignments)]
+            let mut hedge_tile = HedgeTile::Grass;
+            let mut previous_mid_segment = None;
+            for i in 1..13 {
+                (hedge_tile, previous_mid_segment) =
+                    Self::classify_hedge_segment(&hedge_mask[i - 1..i + 2], previous_mid_segment);
+                if hedge_tile != HedgeTile::Grass {
+                    children.push(Child::Hedge(Hedge::new(
+                        hedge_tile,
+                        hedge_row,
+                        Position::new(i as i32 * 40 - 20, 0),
+                    )));
+                }
+            }
+        }
+
+        Self {
+            y,
+            index,
+            hedge_row,
+            hedge_mask,
+            children,
+        }
+    }
+
+    pub fn without_hedge(index: i32, y: i32) -> Self {
+        Self::new(None, HedgeRow::None, index, y)
+    }
+
+    pub fn classify_hedge_segment(
+        mask_window: &[HedgeMask],
+        previous_mid_segment: Option<HedgeTile>,
+    ) -> (HedgeTile, Option<HedgeTile>) {
+        if mask_window[1] == HedgeMask::Empty {
+            (HedgeTile::Grass, None)
+        } else if mask_window[0] == HedgeMask::Empty && mask_window[2] == HedgeMask::Empty {
+            (HedgeTile::SingleWidth, None)
+        } else if mask_window[0] == HedgeMask::Empty {
+            (HedgeTile::LeftMost, None)
+        } else if mask_window[2] == HedgeMask::Empty {
+            (HedgeTile::RightMost, None)
+        } else {
+            match previous_mid_segment {
+                Some(HedgeTile::Middle4) if mask_window[2] == HedgeMask::Empty => {
+                    (HedgeTile::Middle5, None)
+                }
+                Some(HedgeTile::Middle4) if mask_window[2] == HedgeMask::Hedge => {
+                    (HedgeTile::Middle3, Some(HedgeTile::Middle3))
+                }
+                Some(HedgeTile::Middle3) => (HedgeTile::Middle3, Some(HedgeTile::Middle3)),
+                _ => (HedgeTile::Middle3, Some(HedgeTile::Middle3)),
+            }
+        }
+    }
+
+    pub fn first_hedge_row(index: i32) -> (Vec<HedgeMask>, HedgeRow) {
+        if rand::gen_range::<u8>(0, 1) == 0 && index > 7 && index < 14 {
+            (Self::generate_hedge_mask(), HedgeRow::First)
+        } else {
+            (Vec::new(), HedgeRow::None)
+        }
+    }
+
+    pub fn generate_hedge_mask() -> Vec<HedgeMask> {
+        let mut mask = Vec::new();
+        mask.resize_with(12, || {
+            if rand::gen_range::<u8>(1, 100) > 1 {
+                HedgeMask::Hedge
+            } else {
+                HedgeMask::Empty
+            }
+        });
+        // Ensure there is at least one gap
+        mask.insert(rand::gen_range(0, 11), HedgeMask::Empty);
```

Comment:
In the source, the new element is set, while here it's added - not sure if this is intentional :thinking:.

This is a difference that doesn't impact the game in a noticeable way, but it's curious :smile:

Reply:
Figuring out what this part did and trying to implement it in Rust was one of the trickiest parts of this port. So I’m not surprised there is a bug. I can’t remember if it was intentional or not. I’ll take a a look at it.